src/ctap/mod.rs

```diff
@@ -280,22 +284,40 @@ impl StatefulPermission {
         StatefulPermission {
             permission: TimedPermission::granted(now, RESET_TIMEOUT_DURATION),
             command_type: Some(StatefulCommand::Reset),
+            channel: None,
         }
     }
 
     /// Clears all permissions and state.
     pub fn clear(&mut self) {
         self.permission = TimedPermission::waiting();
         self.command_type = None;
+        self.channel = None;
+    }
+
+    /// Clears all state if communication is coming from a different channel.
+    pub fn clear_old_channels(&mut self, channel: &Channel) {
+        // There are different possible choices for incoming traffic on a different channel:
+        // A) Always reset state (our choice).
+        // B) Only reset state if the new command is stateful.
+        // C) Keep state on all channels until timeout.
+        //
+        // If we wanted to switch to (B) or (C), we'd have to be very careful that the state does
+        // not go stale. For example, we keep credential keys that we expect to still exist.
+        // However, interleaving (stateless) commands could delete credentials or change the PIN,
+        // which chould make invalidate our access. Some read-only commands should be okay to run,
+        // but (A) is the safest and easiest solution.
+        if let Some(c) = self.channel.as_ref() {
```

Comment:
nit/fyi: `&self.channel` should work too

Reply:
Addressed in #449.